mypy/semanal.py

```diff
@@ -4310,3 +4334,21 @@ def find_fixed_callable_return(expr: Expression) -> Optional[CallableType]:
             if isinstance(t.ret_type, CallableType):
                 return t.ret_type
     return None
+
+
+def unmark_any_as_explicit(t: Type) -> None:
+    """For all types within t, if that type is Any, set explicit to False"""
+    t.accept(UnmarkAnyAsExplicit())
+
+
+class UnmarkAnyAsExplicit(TypeQuery[None]):
```

Comment:
Yeah, I agree, this does not seem right.
Moreover, this mutates a type, which is also not great.

What I need is a TypeVisitor that visit all types within a type. Maybe I can refactor one from `TypeQuery` and then make `TypeQuery` inherit the new visitor or something?